modules/nextflow/src/main/groovy/nextflow/k8s/model/JobSpecBuilder.groovy

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020-2022, Seqera Labs
+ * Copyright 2013-2019, Centre for Genomic Regulation (CRG)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.k8s.model
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+
+/**
+ * Object build for a K8s job specification
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@CompileStatic
+@Slf4j
+class JobSpecBuilder {
+
+    String jobName
+
+    String namespace
+
+    String workDir
+
+    int backoffLimit = -1
```

Comment:
It would be better to use `Integer` and check for `null` instead of `-1` as absent value.

Reply:
Can you give an advice how to distinguish between `null` and `0` for `Integer` object?

Reply:
Having `Integer backoffLimit` you can check for nullability with `backoffLimit == null`

Reply:
at first, we wanted to keep default behaviour of jobs. If backofflimit is not set, it has a default value of 6. However, as restarting pods in place is not desired modus operandi of nextflow, we make it simplier just always setting backoffLimit to 0 unless explicitly specified by withJobBackoffLimit() method. Is this approach ok?

Reply:
If so, I agree, that it would be better to default to `0` and delegate the retry to nextflow

Reply:
ok, resolved then.